### PR TITLE
feat: preview of installed app name on the schedule screen

### LINKED ISSUE #12689: [CAL-2932] feat: Improvement preview of installed app on the schedule screen 🚨
### Is your proposal related to a problem?

Currently, whenever we select any of the installed apps (Example: Discord/Telegram) (i.e., I1), and provided URL is shown as given in the setting (i.e, I2).

---

* I1:

image

* I2:

image

---

### Describe the solution you'd like

In this case, instead of showing the URL as it is when preview enabled in settings, we can try to hyperlink the app name rather than providing the URL directly. That would be a much cleaner UX.

### Describe alternatives you've considered

NA

### Additional context

NA

### Requirement/Document

To preview the App name hyper linked instead of showing the URL as it is.

---

##### House rules

* If this issue has a `🚨 needs approval` label, don't start coding yet. Wait until a core member approves feature request by removing this label, then you can start coding.
  * For clarity: Non-core member issues automatically get the `🚨 needs approval` label.
  * Your feature ideas are invaluable to us! However, they undergo review to ensure alignment with the product's direction.

<sub>From [SyncLinear.com](https://synclinear.com) | [CAL-2932](https://linear.app/calcom/issue/CAL-2932/feat-improvement-preview-of-installed-app-on-the-schedule-screen-🚨)</sub>

### PR DESCRIPTION
**Pull request by zhravan**
## What does this PR do?

Fixes  https://github.com/calcom/cal.com/issues/12689

## Requirement/Documentation

image

----
----

Output after the change has been made:

image

## Type of change

- Bug fix (non-breaking change which fixes an issue)

## Mandatory Tasks

- [ x ] Make sure you have self-reviewed the code. A decent size PR without self-review might be rejected.

## Checklist

- I haven't read the [contributing guide](https://github.com/calcom/cal.com/blob/main/CONTRIBUTING.md)
- My code doesn't follow the style guidelines of this project
- I haven't commented my code, particularly in hard-to-understand areas
- I haven't checked if my PR needs changes to the documentation
- I haven't checked if my changes generate no new warnings
- I haven't added tests that prove my fix is effective or that my feature works
- I haven't checked if new and existing unit tests pass locally with my changes

### CODE REVIEW

File: packages/features/bookings/Booker/components/BookEventForm/BookingFields.tsx [hunk 1]
```diff
@@ -105,10 +112,20 @@ export const BookingFields = ({
...
+          field.options.map((option) => {
+            option.label =
+              option.label.search(/^https?:/) !== -1 ? (
+                <a href={option.label} target="_blank">
+                  {getDomain(option.label)}{" "}
+                </a>
+              ) : (
+                option.label
+              );
+          });
```
Comment (Udit-takkar) [inline packages/features/bookings/Booker/components/BookEventForm/BookingFields.tsx:128]: type error in CI

Comment (shravan20) [reply packages/features/bookings/Booker/components/BookEventForm/BookingFields.tsx:128]: @Udit-takkar: Can I change the type for  option.label to `string | HTMLElement`?

**Review by Udit-takkar — CHANGES_REQUESTED** [review]
@shravan20 you can probably make changes here https://github.com/calcom/cal.com/blob/d918f34ae8475db6682b01ec5d685861c2b61776/packages/features/form-builder/Components.tsx#L422 and make this a hyperlink when option.value is url. 

make sure to test everything that you can click on links and underline them.

**Comment by shravan20** [comment]
@Udit-takkar: Adding the design test and dry run result for reference 

👉🏻 https://www.awesomescreenshot.com/video/24092331

Re-requesting you for review again. 

Thanks in advance for your time on the review!

**Review by Udit-takkar — APPROVED** [review]

**Comment by Udit-takkar** [comment]
@shravan20 I have pushed some changes. could you sign CLA https://github.com/calcom/cal.com/pull/13221#issuecomment-1891342745 so we can merge it ?

**Comment by shravan20** [comment]
> @shravan20 I have pushed some changes. could you sign CLA [#13221 (comment)](https://github.com/calcom/cal.com/pull/13221#issuecomment-1891342745) so we can merge it ?

@Udit-takkar: I have signed the CLA.

**Comment by shravan20** [comment]
This is good to merge, thanks. :)

**Pull request merged**